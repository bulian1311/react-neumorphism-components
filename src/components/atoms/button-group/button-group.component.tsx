import React, { FC } from 'react';
import { StyledGroup } from './button-group.styled';

export const ButtonGroup: FC = ({ children }) => {
  return <StyledGroup>{children}</StyledGroup>;
};

ButtonGroup.displayName = 'ButtonGroup';

ButtonGroup.propTypes = {};

ButtonGroup.defaultProps = {};
