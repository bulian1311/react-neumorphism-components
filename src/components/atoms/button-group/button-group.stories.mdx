import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as stories from './button-group.stories.tsx';
import { Button } from '../button/button.component.tsx';

<Meta title="Atoms/ButtonGroup" component={Button.Group} />

# ButtonGroup

If you want to group buttons together on a single line, use `<Button.Group` modifier as container.

<ArgsTable of={Button.Group} />

## Example

<Canvas>
  <Story story={stories.ButtonGroup} name="ButtonGroup" />
</Canvas>
