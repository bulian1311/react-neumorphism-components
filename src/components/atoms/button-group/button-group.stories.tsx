import React from 'react';
import { Story } from '@storybook/react';

import { Button } from '../button';

const Template: Story = () => (
  <Button.Group>
    <Button style={{ marginRight: 20 }}>Button 1</Button>
    <Button style={{ marginRight: 20 }}>Button 2</Button>
    <Button>Button 3</Button>
  </Button.Group>
);

export const ButtonGroup = Template.bind({});

ButtonGroup.args = {};

ButtonGroup.argTypes = {};
