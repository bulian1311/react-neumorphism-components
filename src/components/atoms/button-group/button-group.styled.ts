import styled from 'styled-components';

export const StyledGroup = styled.div`
  display: flex;
  justify-content: flex-start;
`;
