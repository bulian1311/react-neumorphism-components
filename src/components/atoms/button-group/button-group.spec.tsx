import React from 'react';
import { ButtonGroup } from './button-group.component';
import { render } from '@testing-library/react';

describe('Box', () => {
  it('matches snapshot', () => {
    const { container } = render(<ButtonGroup>Box</ButtonGroup>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
