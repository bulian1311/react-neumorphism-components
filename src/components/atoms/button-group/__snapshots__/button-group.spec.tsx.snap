// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box matches snapshot 1`] = `
<div
  class="sc-bdvvtL iOPOkm"
>
  Box
</div>
`;
